### PR TITLE
Create docs-elastic-staging-publish.yml

### PR DESCRIPTION
**Pull request by scottybollinger**
In pursuit of https://github.com/elastic/docsmobile/issues/378

I have a TODO to update the README to remove the project names and secrets from the template code after we remove the required attribute and hard-code the name in the workflows.

### CODE REVIEW

File: .github/workflows/docs-elastic-staging-publish.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        with:
+          repository: elastic/${{ inputs.repo }}
+          token: ${{ secrets.VERCEL_GITHUB_TOKEN }}
+          path: ${{ github.workspace }}/${{ inputs.repo }}
+          persist-credentials: false
+
+      - name: Checkout Wordlake-staging
+        uses: actions/checkout@v3.0.1
+        with:
+          repository: elastic/${{ inputs.prebuild }}
```
Comment (scottybollinger) [inline .github/workflows/docs-elastic-staging-publish.yml:67]: This relies on an `input`. What are the ramifications for the migration to staging and then the promotion to docs.elastic.co?

File: .github/workflows/docs-elastic-staging-publish.yml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+          persist-credentials: false
+
+      - name: Prepare content for deploy
+        if: ${{ github.event.pull_request.merged }}
+        uses: actions/checkout@v3.0.1
+        with:
+          path: 'tmp'
+          persist-credentials: false
+
+      - name: Checkout dev doc docsmobile app
```
Comment (glitteringkatie) [inline .github/workflows/docs-elastic-staging-publish.yml:56]:
```suggestion
      - name: Checkout staging site repo
```

File: README.md [hunk 3]
```diff
@@ -122,3 +122,48 @@ jobs:
       VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
       VERCEL_PROJECT_ID_DOCS_CO: ${{ secrets.VERCEL_PROJECT_ID_DOCS_CO }}
 ```
+
+
+### Staging docs builder, calling workflow
+
+Change values as needed.
+
+For example, if you do not have a docs dir, use the correct dir or no dir instead.
```
Comment (glitteringkatie) [inline README.md:131]:
```suggestion
```

We no longer default to having the docs dir in front of things so I don't think this is necessary

**Review by glitteringkatie — CHANGES_REQUESTED** [review]
Two nits but this looks good!

**Review by glitteringkatie — APPROVED** [review]
Yay!

**Pull request merged**